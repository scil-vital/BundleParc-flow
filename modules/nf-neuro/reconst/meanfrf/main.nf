

process RECONST_MEANFRF {
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
        tuple val(prefix), path(frf_list)

    output:
        tuple val(prefix), path("*mean_frf.txt")         , emit: meanfrf
        path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    """
    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
    export OMP_NUM_THREADS=1
    export OPENBLAS_NUM_THREADS=1

    scil_frf_mean $frf_list ${prefix}_mean_frf.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    """
    scil_frf_mean -h

    touch ${prefix}_mean_frf.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
