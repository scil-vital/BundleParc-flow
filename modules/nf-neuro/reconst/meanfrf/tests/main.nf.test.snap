{
    "reconst - meanfrf": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "test_mean_frf.txt:md5,70e339a7ede1f4605c91fe80b150966f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b552ca0c54b99e4f5e26e5b1af5b0811"
                ],
                "meanfrf": [
                    [
                        "test",
                        "test_mean_frf.txt:md5,70e339a7ede1f4605c91fe80b150966f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b552ca0c54b99e4f5e26e5b1af5b0811"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-22T21:02:48.243194681"
    }
}